### PR TITLE
Refactor to use additional classes

### PR DESCRIPTION
**Pull request by malandrina**
- Refactor to use additional classes, WordReferenceApi and
  ResponseParser, to handle querying API and parsing response.
- Upgrade rspec and add rspec-mocks
- Remove whitespace and newlines

### CODE REVIEW

File: lib/word_reference/dictionary.rb [hunk 1]
```diff
@@ -22,29 +20,28 @@ def change_language(dictionary_name)
...
+      response_parser.run
     end
 
     private
 
-    # Configures each setting for API call
     def config
       WordReference::Configurable.keys.each do |key|
-        instance_variable_set(:"@#{key}", WordReference.instance_variable_get(:"@#{key}"))
+        instance_variable_set(
```
Comment (sandbochs) [inline lib/word_reference/dictionary.rb:32]: Nice

**Comment by sandbochs** [comment]
Looks awesome. The response parser class is a great addition.

**Comment by malandrina** [comment]
Thanks @sandbochs! I added a commit that actually fixes one of the bugs: https://github.com/malandrina/word_reference/commit/366ec6e47ae59efa26a25a5146ac671dca0f6756. Can you take a look when you have a moment?

**Comment by malandrina** [comment]
@sandbochs yes I agree -- when I was reviewing the `Dictionary`, `Query`, and `Translation` classes I had a hard time figuring out exactly what they did! Perhaps some additional refactoring on the private interfaces will help.

For now, I'll go ahead and merge this in.

**Pull request merged**